app/src/main/java/com/sunasterisk/weather/data/source/local/database/DBHelper.kt

```diff
@@ -0,0 +1,275 @@
+
+class DBHelper(
+    private val context: Context?
+) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
+
+    override fun onCreate(db: SQLiteDatabase?) {
+        db?.apply {
+            execSQL("CREATE TABLE $TABLE_WEATHER(" +
+                "$COLUMN_ID_WEATHER TEXT PRIMARY KEY, " +
+                "${WeatherEntry.LATITUDE} REAL, " +
+                "${WeatherEntry.LONGITUDE} REAL)"
+            )
+            execSQL("CREATE TABLE $TABLE_CURRENT(" +
+                "${WeatherStatisticsEntry.TIME} INTEGER PRIMARY KEY, " +
+                "${WeatherStatisticsEntry.ICON} CHAR(10), " +
+                "${WeatherStatisticsEntry.SUMMARY} TEXT, " +
+                "${WeatherStatisticsEntry.HUMIDITY} REAL, " +
+                "${WeatherStatisticsEntry.TEMPERATURE} REAL, " +
+                "${WeatherStatisticsEntry.TEMPERATURE_MIN} REAL, " +
+                "${WeatherStatisticsEntry.TEMPERATURE_MAX} REAL, " +
+                "${WindEntry.WIND_SPEED} REAL, " +
+                "${WindEntry.WIND_DIRECTION} INTEGER, " +
+                "$COLUMN_ID_WEATHER TEXT," +
+                "FOREIGN KEY($COLUMN_ID_WEATHER) REFERENCES $TABLE_WEATHER($COLUMN_ID_WEATHER))"
+            )
+            execSQL("CREATE TABLE $TABLE_HOURLY(" +
+                "${WeatherStatisticsEntry.TIME} INTEGER PRIMARY KEY, " +
+                "${WeatherStatisticsEntry.ICON} CHAR(10), " +
+                "${WeatherStatisticsEntry.TEMPERATURE} REAL," +
+                "$COLUMN_ID_WEATHER TEXT," +
+                "FOREIGN KEY($COLUMN_ID_WEATHER) REFERENCES $TABLE_WEATHER($COLUMN_ID_WEATHER))"
+            )
+            execSQL("CREATE TABLE $TABLE_DAILY(" +
+                "${WeatherStatisticsEntry.TIME} INTEGER PRIMARY KEY, " +
+                "${WeatherStatisticsEntry.ICON} CHAR(10), " +
+                "${WeatherStatisticsEntry.TEMPERATURE_MIN} REAL, " +
+                "${WeatherStatisticsEntry.TEMPERATURE_MAX} REAL, " +
+                "$COLUMN_ID_WEATHER TEXT," +
+                "FOREIGN KEY($COLUMN_ID_WEATHER) REFERENCES $TABLE_WEATHER($COLUMN_ID_WEATHER))"
+            )
+        }
+    }
+
+    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
+        db?.apply {
+            execSQL("DROP TABLE IF EXISTS $TABLE_CURRENT")
+            execSQL("DROP TABLE IF EXISTS $TABLE_HOURLY")
+            execSQL("DROP TABLE IF EXISTS $TABLE_DAILY")
+            execSQL("DROP TABLE IF EXISTS $TABLE_WEATHER")
+            onCreate(this)
+        }
+    }
+
+    fun insertWeather(weather: Weather) {
+        weather.latitude?.let { latitude ->
+            weather.longitude?.let { longitude ->
+                val idWeather =
+                    WeatherUtils.formatNameLocation(context, latitude, longitude)
+                val cursorWeather = readableDatabase.rawQuery(
+                    "SELECT * FROM $TABLE_WEATHER WHERE $COLUMN_ID_WEATHER = '$idWeather'",
+                    null)
```

Comment:
sử dụng hàm của SQLiteOpenHelper
chỉ dùng rawQuery với câu lệnh thực sự phức tạp 
```suggestion
                val cursorWeather = readableDatabase.query(...)
```